core/commands/remotepin.go

```diff
@@ -0,0 +1,515 @@
+		defer cancel()
+
+		psCh, errCh, err := lsRemote(ctx, req, env)
+		if err != nil {
+			return err
+		}
+
+		for ps := range psCh {
+			if err := res.Emit(&RemotePinOutput{
+				RequestID: ps.GetRequestId(),
+				Name:      ps.GetPin().GetName(),
+				Delegates: multiaddrsToStrings(ps.GetDelegates()),
+				Status:    ps.GetStatus().String(),
+				Cid:       ps.GetPin().GetCid().String(),
+			}); err != nil {
+				return err
+			}
+		}
+
+		return <-errCh
+	},
+	Type: RemotePinOutput{},
+	Encoders: cmds.EncoderMap{
+		cmds.Text: cmds.MakeTypedEncoder(func(req *cmds.Request, w io.Writer, out *RemotePinOutput) error {
+			fmt.Printf("pin_id=%v\n", out.RequestID)
+			fmt.Printf("pin_name=%q\n", out.Name)
+			for _, d := range out.Delegates {
+				fmt.Printf("pin_delegate=%v\n", d)
+			}
+			fmt.Printf("pin_status=%v\n", out.Status)
+			fmt.Printf("pin_cid=%v\n", out.Cid)
+			return nil
+		}),
+	},
+}
+
+func lsRemote(ctx context.Context, req *cmds.Request, env cmds.Environment) (chan pinclient.PinStatusGetter, chan error, error) {
+	opts := []pinclient.LsOption{}
+	if name, nameFound := req.Options[pinNameOptionName].(string); nameFound {
+		opts = append(opts, pinclient.PinOpts.FilterName(name))
+	}
+	if cidsRaw, cidsFound := req.Options[pinCIDsOptionName].([]string); cidsFound {
+		parsedCIDs := []cid.Cid{}
+		for _, rawCID := range cidsRaw {
+			parsedCID, err := cid.Decode(rawCID)
+			if err != nil {
+				return nil, nil, fmt.Errorf("CID %s cannot be parsed (%v)", rawCID, err)
+			}
+			parsedCIDs = append(parsedCIDs, parsedCID)
+		}
+		opts = append(opts, pinclient.PinOpts.FilterCIDs(parsedCIDs...))
+	}
+	if statusRaw, statusFound := req.Options[pinStatusOptionName].([]string); statusFound {
+		parsedStatuses := []pinclient.Status{}
+		for _, rawStatus := range statusRaw {
+			if pinclient.Status(rawStatus).String() == string(pinclient.StatusUnknown) {
+				return nil, nil, fmt.Errorf("status %s is not valid", rawStatus)
+			}
+			parsedStatuses = append(parsedStatuses, pinclient.Status(rawStatus))
+		}
+		opts = append(opts, pinclient.PinOpts.FilterStatus(parsedStatuses...))
+	}
+
+	service, _ := req.Options[pinServiceNameOptionName].(string)
+	url, key, err := getRemotePinServiceOrEnv(env, service)
+	if err != nil {
+		return nil, nil, err
+	}
+	c := pinclient.NewClient(url, key)
+
+	psCh, errCh := c.Ls(ctx, opts...)
+
+	return psCh, errCh, nil
+}
+
+var rmRemotePinCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Remove pinned objects from remote pinning service.",
+		ShortDescription: `
+Removes the pin from the given object allowing it to be garbage
+collected if needed.
+`,
+	},
+
+	Arguments: []cmds.Argument{
+		cmds.StringArg("request-id", false, true, "Request ID of the pin to be removed.").EnableStdin(),
+	},
+	Options: []cmds.Option{
+		cmds.StringOption(pinNameOptionName, "Remove pins objects with names that contain provided value (case-insensitive, partial or full match)."),
```

Comment:
We switched to `exact` by default, so this can be simplified:
```suggestion
		cmds.StringOption(pinNameOptionName, "Remove pins objects with names that contain provided value."),
```